e2e-tests/src/tests/scheduler.test.ts

```diff
@@ -184,18 +190,87 @@ test.describe('Scheduling', () => {
       reason_local = reason;
       expect(status).not.toBeNull();
       expect(status).toBeDefined();
+      if(datasetId != null){
+        dataset_id = datasetId
+      }
       await delay(1000);
     }
     if (status_local == "failed") {
       throw new Error(reason_local);
     }
     expect(status_local).toEqual("complete")
     expect(analysisId_local).toEqual(analysisId)
+    expect(dataset_id).not.toBeNull();
+  });
+
+  test('Verify posting of simulation results', async ({ request }) => {
+    let plan = await req.getPlan(request, plan_id)
+    let directiveIds = new Set<number>()
+    let startOffsetsActivityDirectives = new Set<string>()
+    for(const activity of plan.activity_directives){
+      directiveIds.add(activity['id'])
+    }
+    const activities = await req.getSimulationDatasetByDatasetId(request,dataset_id)
+    expect(activities.simulated_activities.length).toEqual(plan.activity_directives.length)
+    let refDirectiveIds = new Set<number>()
+    for(let simulated_activity of activities.simulated_activities){
+      refDirectiveIds.add(simulated_activity.activity_directive.id)
+      startOffsetsActivityDirectives.add(simulated_activity.start_offset)
+    }
+    //all directive have their simulated activity
+    expect(eqSet(refDirectiveIds,directiveIds))
+    const dataset = await req.getProfiles(request, dataset_id)
+    //expect one profile per resource in the banananation model
+    expect(dataset.length).toEqual(7)
+    for(let resource of dataset){
+      if(resource.name =="/fruit" || resource.name == "/peel"){
+        let startOffsetOfResource = new Set<string>()
+        for(let segment of resource.profile_segments){
+          startOffsetOfResource.add(segment.start_offset)
+        }
+        //these two resources are affected by the peel, grow and bite banana
+        expect(eqSet(startOffsetOfResource, startOffsetsActivityDirectives)).toEqual(true)
+      }
+    }
+    let topics = await req.getTopicsEvents(request, dataset_id)
+    let prefixInput = "ActivityType.Input."
+    let prefixOutput = "ActivityType.Output."
+    let peelIsThere = false
+    let biteIsThere = false
+    let growIsThere = false
+    for(let topic of topics){
+      switch (topic.name){
+        case prefixInput+"BiteBanana":
+          biteIsThere = true
+          expect(topic.events.length).toEqual(1)
+          break
+        case prefixOutput+"BiteBanana":
+          expect(topic.events.length).toEqual(1)
+          break
+        case prefixInput+"GrowBanana":
+          growIsThere = true
+          expect(topic.events.length).toEqual(1)
+          break
+        case prefixOutput+"GrowBanana":
+          expect(topic.events.length).toEqual(1)
+          break
+        case prefixInput+"PeelBanana":
+          peelIsThere = true
+          expect(topic.events.length).toEqual(12)
+          break
+        case prefixOutput+"PeelBanana":
+          expect(topic.events.length).toEqual(12)
+          break
+      }
```

Comment:
Regarding this switch:
- Can you add an `expect` for the size of `topic` (looks like it should be 6)?
- Can you add booleans to track that the prefixOutput topics occur as well?

Reply:
1) No, topics can contain more stuff, in this example, it contains also a topic about a resource. And I am not sure exactly why. 
2) What do you mean ? in the switch, there both prefixInput and prefixOutput.

Reply:
Oh I think I got it for the booleans.

Reply:
Oh, if it also is supposed to have a resource topic, then those resource topics should also be checked that they exist. Otherwise, this part of the test isn't asserting that topic contains what it's supposed to.

Reply:
ok I have added the check for the topic.